web_search_nospace/static/src/js/search.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * © 2017 Therp BV <http://therp.nl> License AGPL-3.0 or later
+ * (http://www.gnu.org/licenses/agpl.html).
+ */
+openerp.web_search_nospace = function(instance) {
+	instance.web.search.InputView.include({
+
+		onPaste : function() {
+			this._super()
+			setTimeout(function() {
```

Comment:
I am not an expert in Odoo javascript programming. But is the timeout really needed here? Can't we have simply:  ```this.$el.text(this.$el.text().trim());``` directly? Just asking...

Reply:
@daramousk Please answer this question as well

Reply:
@NL66278 Basically there were some timing/race condition issues from what I remember, you can see an explanation here https://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful.

Reply:
@daramousk Did you actually experience these conditions? Because if I read up on the discussion on stackoverflow, especially the extended answer by DVK, this does not seem to apply here, but maybe I am mistaken.

Reply:
@NL66278 Ok I just went back to this to remember why I used it. Basically I used it because it was used on the super call and for some reason the the changes to `$el` are not propagated unless you force it with setTimeout. If you remove the setTimeout the whole thing will not work at all. 

Now why this happens I do not know, it's probably some obscure event handling mechanism. You can check the super call for more data.

Reply:
OK, lets keep it as you made it.